Source/Core/VideoBackends/D3D12/PixelShaderCache.cpp

```diff
@@ -0,0 +1,651 @@
+// Copyright 2010 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <string>
+
+#include "Common/FileUtil.h"
+#include "Common/LinearDiskCache.h"
+#include "Common/StringUtil.h"
+
+#include "Core/ConfigManager.h"
+
+#include "VideoBackends/D3D12/D3DBase.h"
+#include "VideoBackends/D3D12/D3DCommandListManager.h"
+#include "VideoBackends/D3D12/D3DDescriptorHeapManager.h"
+#include "VideoBackends/D3D12/D3DShader.h"
+#include "VideoBackends/D3D12/Globals.h"
+#include "VideoBackends/D3D12/PixelShaderCache.h"
+
+#include "VideoCommon/Debugger.h"
+#include "VideoCommon/PixelShaderGen.h"
+#include "VideoCommon/PixelShaderManager.h"
+#include "VideoCommon/Statistics.h"
+#include "VideoCommon/VideoConfig.h"
+
+namespace DX12
+{
+
+PixelShaderCache::PSCache PixelShaderCache::PixelShaders;
+const PixelShaderCache::PSCacheEntry* PixelShaderCache::last_entry;
+PixelShaderUid PixelShaderCache::last_uid = {};
+UidChecker<PixelShaderUid,ShaderCode> PixelShaderCache::pixel_uid_checker;
+
+LinearDiskCache<PixelShaderUid, u8> g_ps_disk_cache;
+
+D3DBlob* s_ColorMatrixProgramBlob[2] = {};
+D3DBlob* s_ColorCopyProgramBlob[2] = {};
+D3DBlob* s_DepthMatrixProgramBlob[2] = {};
+D3DBlob* s_ClearProgramBlob = {};
+D3DBlob* s_AnaglyphProgramBlob = {};
+D3DBlob* s_rgba6_to_rgb8Blob[2] = {};
+D3DBlob* s_rgb8_to_rgba6Blob[2] = {};
+
+ID3D12Resource* pscbuf12 = nullptr;
+D3D12_GPU_VIRTUAL_ADDRESS pscbuf12GPUVA = {};
+void* pscbuf12data = nullptr;
+const UINT pscbuf12paddedSize = (sizeof(PixelShaderConstants) + 0xff) & ~0xff;
```

Comment:
These should all (from here and up) be static and have the s_ prefix in the name.